src/Sarif/Core/FileData.cs

```diff
@@ -12,19 +15,40 @@ namespace Microsoft.CodeAnalysis.Sarif
     /// </summary>
     public partial class FileData : ISarifNode
     {
-        public static FileData Create(Uri uri, bool computeHashes)
+        public static FileData Create(Uri uri, SarifWriters.LoggingOptions loggingOptions)
         {
             if (uri == null) { throw new ArgumentNullException(nameof(uri)); }
 
             var fileData = new FileData()
             {
-                MimeType = SarifWriters.MimeType.DetermineFromFileExtension(uri)
+                MimeType = SarifWriters.MimeType.DetermineFromFileExtension(uri),
+                Uri = uri
```

Comment:
>Uri [](start = 16, length = 3)

ouch, we never set the URI before. not sure yet why this wasn't a problem. Unit testing FTW

Reply:
Probably wasn't a problem because in most cases the URI recaps the dictionary key.

---
In reply to: [136142944](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136142944) [](ancestors = 136142944)